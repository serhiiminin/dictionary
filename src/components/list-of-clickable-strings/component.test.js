import React from 'react';
import renderer from 'react-test-renderer';
import ListOfClickableStrings from './component';

describe('List of clickable strings', () => {
  const component = renderer.create(
    <ListOfClickableStrings
      items={['one', 'two']}
      onClick={() => {}}
    />
  );
  const tree = component.toJSON();

  test('render', () => {
    expect(tree).toMatchSnapshot();
  });
});
