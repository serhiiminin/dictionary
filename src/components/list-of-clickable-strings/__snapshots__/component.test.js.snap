// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List of clickable strings render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListOfClickableStrings
    delimiter=""
    items={
        Array [
            "one",
            "two",
          ]
    }
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Jss(ClickableWord)
          delimiter=""
          onClick={[Function]}
          word="one"
/>,
        <Jss(ClickableWord)
          delimiter=""
          onClick={[Function]}
          word="two"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "8fac9fa1-2b19-4a34-95ec-51e17e3cc04d",
        "nodeType": "class",
        "props": Object {
          "delimiter": "",
          "onClick": [Function],
          "word": "one",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "dec4916f-93ca-47cb-8366-997925515038",
        "nodeType": "class",
        "props": Object {
          "delimiter": "",
          "onClick": [Function],
          "word": "two",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Jss(ClickableWord)
            delimiter=""
            onClick={[Function]}
            word="one"
/>,
          <Jss(ClickableWord)
            delimiter=""
            onClick={[Function]}
            word="two"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "8fac9fa1-2b19-4a34-95ec-51e17e3cc04d",
          "nodeType": "class",
          "props": Object {
            "delimiter": "",
            "onClick": [Function],
            "word": "one",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "dec4916f-93ca-47cb-8366-997925515038",
          "nodeType": "class",
          "props": Object {
            "delimiter": "",
            "onClick": [Function],
            "word": "two",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
